resources/prosody-plugins/mod_fmuc.lua

```diff
@@ -0,0 +1,134 @@
+--- activate under main muc component
+--- Add the following config under the main muc component
+---  muc_room_default_presence_broadcast = {
+---        visitor = false;
+---        participant = true;
+---        moderator = true;
+---    };
+--- Enable in global modules: 's2s_bidi'
+--- Make sure 's2s' is not in modules_disabled
+--- TODO: Do we need the /etc/hosts changes? We can drop it for https://modules.prosody.im/mod_s2soutinjection.html
+--- In /etc/hosts add:
+--- vmmain-ip-address focus.domain.com
+--- vmmain-ip-address conference.domain.com
+--- vmmain-ip-address domain.com
+--- Open port 5269 on the provider side and on the firewall of the machine, so the core node can access this visitor one
+local jid = require 'util.jid';
+local st = require 'util.stanza';
+
+local muc_domain_prefix = module:get_option_string('muc_mapper_domain_prefix', 'conference');
+local main_domain = string.gsub(module.host, muc_domain_prefix..'.', '');
+
+local function get_focus_occupant(room)
+    local focus_occupant = room._data.focus_occupant;
+
+    if focus_occupant then
+        return focus_occupant;
+    end
+    if not focus_occupant then
+        for _, n_occupant in room:each_occupant() do
+            if jid.node(n_occupant.jid) == 'focus' then
```

Comment:
We recognize jicofo based on the nickname. Is it possible for users to login with this nickname?

Reply:
Nope, as jicofo is the first one to join, only jicofo can create the rooms.